src/evidently/utils/data_preprocessing.py

```diff
@@ -510,12 +552,23 @@ def _get_column_presence(column_name: str, data: _InputData) -> ColumnPresenceSt
     return ColumnPresenceState.Partially
 
 
+def _get_column_cardinality(column_name: str, data: _InputData) -> float:
+    if column_name in data.current.columns and data.current.shape[0] > 0:
+        count = data.current[column_name].nunique()
+        return count
+    return 0
+
+
 NUMBER_UNIQUE_AS_CATEGORICAL = 5
 
 
-def _get_column_type(column_name: str, data: _InputData, mapping: Optional[ColumnMapping] = None) -> ColumnType:
+def _get_column_type(
+    column_name: str, data: _InputData, mapping: Optional[ColumnMapping] = None, cardinality_limit: Optional[int] = None
+) -> ColumnType:
     if mapping is not None:
         if mapping.categorical_features and column_name in mapping.categorical_features:
+            if cardinality_limit and _is_cardinality_exceeded(column_name, data, cardinality_limit):
```

Comment:
Better reuse result of `_is_cardinality_exceeded` as it is expensive call.

Reply:
I use _is_cardinality_exceeded twice in _get_column_type but in different "if" branches, so _is_cardinality_exceeded never will be called twice.